jvb/src/main/kotlin/org/jitsi/videobridge/EndpointK.kt

```diff
@@ -84,9 +85,9 @@ class EndpointK @JvmOverloads constructor(
         })
     }
 
-    private val bandwidthProbing = BandwidthProbing(
-        _transceiver::sendProbing,
-        bitrateController::getStatusSnapshot
+    private val bandwidthProbing = BandwidthProbingK(
+        BandwidthProbing.ProbingDataSender { mediaSsrcs, numBytes -> _transceiver.sendProbing(mediaSsrcs, numBytes) },
+        java.util.function.Supplier { bitrateController.getStatusSnapshot() }
```

Comment:
Now that this is Kotlin can it use a Kotlin idiom here rather than `java.util.function.Supplier`?

Reply:
I actually think `Supplier` still works pretty well here.  If we change it, `BandwidthProbing` would just take: `() -> BitrateControllerStatusSnapshot`, but I find the naming gets a little awkward.  Currently the member is `statusSnapshotSupplier` and we call `statusSnapshotSupplier.get()`, if it was just a function it would just be `statusSnapshotSupplier`, so I think we'd want to rename the member to something like `getStatusSnapshot` which looks better when called, but is a bit weird when passing it in.  Either way I don't think I feel strongly so I can change it if you want.

Reply:
Ok, this is probably fine then.

Is this something that Kotlin 1.4's changes would improve?

Reply:
I don't think this would change, because SAM conversion for _Java_ interfaces is already supported it's the support for Kotlin interfaces that isn't in yet.  Do you just not like the `java.util.function.Supplier` part?  I just realized I can get rid of the package name, so it'll just be `Supplier { ... }` at least.

Reply:
It's mostly just that it was wordier than the equivalent Java code, which tends to annoy me.

Yeah, losing the package name would help.